### PR TITLE
[IMP] test_discuss_full: add reactions to TestDiscussFullPerformance

### PR DESCRIPTION
**Pull request by maki-odoo**
Preparation for PR #170553

### CODE REVIEW

File: addons/test_discuss_full/tests/test_performance.py [hunk 1]
```diff
@@ -16,18 +16,19 @@ class TestDiscussFullPerformance(HttpCase, MailCommon):
     #     1: hasGifPickerFeature
     #     1: hasLinkPreviewFeature
     #     1: hasMessageTranslationFeature
+    #     1: authentication to add reaction
```
[the remark this answers is not in the document]
Comment (maki-odoo) [reply addons/test_discuss_full/tests/test_performance.py:19]:
```
query: SELECT "res_users"."id", "res_users"."partner_id", "res_users"."login", "res_users"."signature", "res_users"."active", "res_users"."action_id", "res_users"."share", "res_users"."company_id", "res_users"."create_uid", "res_users"."create_date", "res_users"."write_uid", "res_users"."write_date", "res_users"."notification_type", "res_users"."karma", "res_users"."rank_id", "res_users"."next_rank_id", "res_users"."odoobot_state", "res_users"."odoobot_failed", "res_users"."sale_team_id", "res_users"."last_seen_phone_call", "res_users"."target_sales_won", "res_users"."target_sales_done", "res_users"."helpdesk_target_closed", "res_users"."helpdesk_target_rating", "res_users"."helpdesk_target_success", "res_users"."website_id" FROM "res_users" WHERE ("res_users"."id" IN (68))
query: SELECT "res_partner"."id", "res_partner"."website_meta_og_img", "res_partner"."seo_name"->>'en_US', "res_partner"."website_id", "res_partner"."is_published", "res_partner"."email_normalized", "res_partner"."message_bounce", "res_partner"."name", "res_partner"."complete_name", "res_partner"."title", "res_partner"."parent_id", "res_partner"."ref", "res_partner"."lang", "res_partner"."tz", "res_partner"."user_id", "res_partner"."vat", "res_partner"."company_registry", "res_partner"."website", "res_partner"."comment", "res_partner"."active", "res_partner"."employee", "res_partner"."function", "res_partner"."type", "res_partner"."street", "res_partner"."street2", "res_partner"."zip", "res_partner"."city", "res_partner"."state_id", "res_partner"."country_id", "res_partner"."partner_latitude", "res_partner"."partner_longitude", "res_partner"."email", "res_partner"."phone", "res_partner"."mobile", "res_partner"."is_company", "res_partner"."industry_id", "res_partner"."company_id", "res_partner"."color", "res_partner"."partner_share", "res_partner"."commercial_partner_id", "res_partner"."commercial_company_name", "res_partner"."company_name", "res_partner"."create_uid", "res_partner"."create_date", "res_partner"."write_uid", "res_partner"."write_date", "res_partner"."contact_address_complete", "res_partner"."signup_type", "res_partner"."calendar_last_notif_ack", "res_partner"."plan_to_change_car", "res_partner"."plan_to_change_bike", "res_partner"."ocn_token", "res_partner"."partner_gid", "res_partner"."additional_info", "res_partner"."phone_sanitized", "res_partner"."debit_limit", "res_partner"."last_time_entries_checked", "res_partner"."invoice_warn", "res_partner"."invoice_warn_msg", "res_partner"."supplier_rank", "res_partner"."customer_rank", "res_partner"."autopost_bills", "res_partner"."ubl_cii_format", "res_partner"."peppol_endpoint", "res_partner"."peppol_eas", "res_partner"."website_description"->>'en_US', "res_partner"."website_short_description"->>'en_US', "res_partner"."avalara_partner_code", "res_partner"."followup_reminder_type", "res_partner"."online_partner_information", "res_partner"."box_1099_id" FROM "res_partner" WHERE ("res_partner"."id" IN (132))
query: SELECT "res_partner"."id", "res_partner"."website_meta_og_img", "res_partner"."seo_name"->>'en_US', "res_partner"."website_id", "res_partner"."is_published", "res_partner"."email_normalized", "res_partner"."message_bounce", "res_partner"."name", "res_partner"."complete_name", "res_partner"."title", "res_partner"."parent_id", "res_partner"."ref", "res_partner"."lang", "res_partner"."tz", "res_partner"."user_id", "res_partner"."vat", "res_partner"."company_registry", "res_partner"."website", "res_partner"."comment", "res_partner"."active", "res_partner"."employee", "res_partner"."function", "res_partner"."type", "res_partner"."street", "res_partner"."street2", "res_partner"."zip", "res_partner"."city", "res_partner"."state_id", "res_partner"."country_id", "res_partner"."partner_latitude", "res_partner"."partner_longitude", "res_partner"."email", "res_partner"."phone", "res_partner"."mobile", "res_partner"."is_company", "res_partner"."industry_id", "res_partner"."company_id", "res_partner"."color", "res_partner"."partner_share", "res_partner"."commercial_partner_id", "res_partner"."commercial_company_name", "res_partner"."company_name", "res_partner"."create_uid", "res_partner"."create_date", "res_partner"."write_uid", "res_partner"."write_date", "res_partner"."contact_address_complete", "res_partner"."signup_type", "res_partner"."calendar_last_notif_ack", "res_partner"."plan_to_change_car", "res_partner"."plan_to_change_bike", "res_partner"."ocn_token", "res_partner"."partner_gid", "res_partner"."additional_info", "res_partner"."phone_sanitized", "res_partner"."debit_limit", "res_partner"."last_time_entries_checked", "res_partner"."invoice_warn", "res_partner"."invoice_warn_msg", "res_partner"."supplier_rank", "res_partner"."customer_rank", "res_partner"."autopost_bills", "res_partner"."ubl_cii_format", "res_partner"."peppol_endpoint", "res_partner"."peppol_eas", "res_partner"."website_description"->>'en_US', "res_partner"."website_short_description"->>'en_US', "res_partner"."avalara_partner_code", "res_partner"."followup_reminder_type", "res_partner"."online_partner_information", "res_partner"."box_1099_id" FROM "res_partner" WHERE ("res_partner"."id" IN (132))
```
These queries are different lines in log files before and after. The line that produce this difference is:
```Py
self.authenticate(self.users[0].login, self.password)
```
And two last queries are the same. I see your point but it seems it tries to read login for the user in the first query. How can I be sure about which is the real cause?

The logs are produced by:
```py
with self.assertQueryCount(emp=self._query_count_init_store):
    import logging
    logging.getLogger('odoo.sql_db').setLevel('DEBUG')
    self.env["res.users"].with_user(self.users[0])._init_store_data(store)
    logging.getLogger('odoo.sql_db').setLevel('INFO')
```

[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/test_discuss_full/tests/test_performance.py:19]: In `sql_db` you can add a conditional breakpoint in `execute` to trigger only on the query that you are investigating, and then check the stack to see where it comes from.

File: addons/test_discuss_full/tests/test_performance.py [hunk 2]
```diff
@@ -260,9 +261,9 @@ def setUp(self):
         }, headers={"Cookie": f"{self.guest._cookie_name}={self.guest._format_auth_cookie()};"})
         # add needaction
         self.users[0].notification_type = 'inbox'
-        message = self.channel_channel_public_1.message_post(body='test', message_type='comment', author_id=self.users[2].partner_id.id, partner_ids=self.users[0].partner_id.ids)
+        message_0 = self.channel_channel_public_1.message_post(body='test', message_type='comment', author_id=self.users[2].partner_id.id, partner_ids=self.users[0].partner_id.ids)
```
Comment (seb-odoo) [inline addons/test_discuss_full/tests/test_performance.py:264]: apply formatter on updated line

File: addons/test_discuss_full/tests/test_performance.py [hunk 3]
```diff
@@ -278,6 +279,30 @@ def setUp(self):
         session = self.env["discuss.channel.rtc.session"].sudo().create(data)
         member_0.rtc_inviting_session_id = session
         self.env.cr.precommit.run()  # trigger the creation of bus.bus records
```
Comment (seb-odoo) [inline addons/test_discuss_full/tests/test_performance.py:281]: This should stay the last line of the setup.

File: addons/test_discuss_full/tests/test_performance.py [hunk 4]
```diff
@@ -1168,6 +1195,64 @@ def _expected_result_for_message(self, channel):
...
+                "notifications": [],
+                "parentMessage": False,
+                "pinned_at": False,
+                "rating_id": False,
+                "reactions": [
+                    {
+                        "content": "👍",
+                        "count": 1,
+                        "personas": [{"id": user_2.partner_id.id, "type": "partner"}],
+                        "message": {"id": last_message.id},
```
Comment (seb-odoo) [inline addons/test_discuss_full/tests/test_performance.py:1223]: sort keys alphabetically in all of those

File: addons/test_discuss_full/tests/test_performance.py [hunk 5]
```diff
@@ -137,8 +138,8 @@ class TestDiscussFullPerformance(HttpCase, MailCommon):
     #       - search mail_tracking_value
     #       - search mail_message_res_partner_starred_rel
     #       - search rating_rating
-    #       6: _compute_rating_stats
-    _query_count_discuss_channels = 56
+    #       7: _compute_rating_stats
```
Comment (seb-odoo) [inline addons/test_discuss_full/tests/test_performance.py:141]: Note: I have an upcoming PR to fix the scaling issue with this.

**Comment by seb-odoo** [comment]
I'll rebase on https://github.com/odoo/odoo/pull/177239 before merging

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**